provider/file/file.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bank-vaults/secret-init/provider"
+)
+
+type Provider struct {
+	SecretsFilePath string
+	SecretData      []byte
+}
+
+func NewFileProvider(secretsFilePath string) (provider.Provider, error) {
+	data, err := os.ReadFile(secretsFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read from file: %w", err)
+	}
+
+	return &Provider{SecretsFilePath: secretsFilePath, SecretData: data}, nil
+}
+
+func (provider *Provider) LoadSecrets(_ context.Context, envs map[string]string) ([]string, error) {
+	// envs that has a "file:" prefix needs to be loaded
+	var secrets []string
+	for key, value := range envs {
+		if strings.HasPrefix(value, "file:") {
+			secret, err := provider.getSecretFromFile(key)
+			if err != nil {
+				return nil, fmt.Errorf("failed to load secret: %w", err)
+			}
+			secrets = append(secrets, fmt.Sprintf("%s=%s", key, secret))
+		}
+	}
```

Comment:
Additional notes:
- Rather than fetching secret data for every secret via `getSecretFromFile`, we can load all secrets from the file to a map using `getSecretsFromFile` instead (you can reuse existing logic)
- We load the secrets in-memory only during the execution of `LoadSecrets` rather than keeping the sensitive file contents in-memory all the time (sidenote: since this is a testing/local provider, safety is not big of a concern, but it's definitely a plus to do it properly)
- We always facilitate runtime file changes (writes/updates/deletes to file) which is not possible if the file is only read once 

Therefore, my suggestion is to rewrite the method to something like this
```suggestion
	// Extract secrets from the file to a map
	secretsMap, err := provider.getSecretsFromFile()
	if err != nil {
		return nil, fmt.Errorf("failed to load secrets: %w", err)
	}
	
	for key, value := range envs {
		if strings.HasPrefix(value, "file:") {
			// TODO: Why do we need "value"? This needs to be clarified first
			
			// Check if the requested secret is in the loaded secret map
			secret, ok := secretsMap[key]
			if !ok {
				return nil, fmt.Errorf("secret %s not found", key)
			}
			secrets = append(secrets, fmt.Sprintf("%s=%s", key, secret))
		}
	}
```

Reply:
Another thing I am failing to see is the relevancy of the actual environment variable value. 
For example, the current flow handles it like this:
- have a secrets file `MYSQL_PASSWORD=3xtr3ms3cr3t` as contents
- have an env variable `MYSQL_PASSWORD=file:secret` (based on the tests)
- `LoadSecrets` will do the injection on `MYSQL_PASSWORD` because:
	- the env variable value starts with `file:`
	- the env variable key `MYSQL_PASSWORD` is found in the secrets file

Therefore:
1. What is the usage of the `secret` part in the env variable `MYSQL_PASSWORD=file:secret`?
2. Why are we using environment variables as keys in the secrets file instead of a specific key we can reference via the env variable?

@sagikazarmark 

### Alternative approach

If we used the following flow, we'd be able to specify file/field via env variable, for example:
- have a YAML secrets file with `sqlPassword: sensitive-value` as contents
- have an env variable `MYSQL_PASSWORD=file:sqlPassword` that references some key from the secrets file (alternatively, we could also reference which secret file to use via `file:/my/secret/file:sqlPassword` but this can be implemented later)
- `LoadSecrets` will do the injection on `MYSQL_PASSWORD` because:
	- the env variable value starts with `file:`
	- the env variable value references a key `sqlPassword` found in the secrets file

Now we have a way to dynamically select a specific secret rather than defining env variables in the secrets file.